gwcs/geometry.py

```diff
@@ -74,29 +81,79 @@ class SphericalToCartesian(Model):
     n_inputs = 2
     n_outputs = 3
 
-    def __init__(self, **kwargs):
+    def __init__(self, wrap_phi_at=360, theta_def='elevation', **kwargs):
+        """
+        Parameters
+        ----------
+        wrap_phi_at : {180, 360}, optional
+            Specifies the range of the azimuthal angle. When ``wrap_phi_at`` is
+            180, azimuthal angle will have a range of ``[-180, 180)`` and
+            when ``wrap_phi_at`` is 360 (default), the azimuthal angle will have
+            a range of ``[0, 360)``
+
+        theta_def : {'elevation', 'inclination', 'polar'}, optional
+            Specifies the definition used for the ``theta``: 'elevation' from
+            the reference plane or 'inclination' (or 'polar') angle measured
+            from the pole.
+
+        """
         super().__init__(**kwargs)
+        self.wrap_phi_at = wrap_phi_at
+        self.theta_def = theta_def
         self.inputs = ('phi', 'theta')
         self.outputs = ('x', 'y', 'z')
 
-    @staticmethod
-    def evaluate(phi, theta):
+    @property
+    def wrap_phi_at(self):
+        return self._wrap_phi_at
+
+    @wrap_phi_at.setter
+    def wrap_phi_at(self, wrap_angle):
+        if not _is_int(wrap_angle):
+            raise TypeError("'wrap_phi_at' must be an integer number: 180 or 360")
+        if wrap_angle not in [180, 360]:
+            raise ValueError("Allowed 'wrap_phi_at' values are 180 and 360")
+        self._wrap_phi_at = wrap_angle
+
+    @property
+    def theta_def(self):
+        return self._theta_def
+
+    @theta_def.setter
+    def theta_def(self, theta_def):
+        if not isinstance(theta_def, str):
+            raise TypeError("'theta_def' must be a string.")
+        if theta_def not in ['elevation', 'inclination', 'polar']:
+            raise ValueError("Allowed 'theta_def' values are: 'elevation', "
+                             "'inclination', or 'polar'.")
+        self._theta_def = theta_def
+
+    def evaluate(self, phi, theta):
         if isinstance(phi, u.Quantity) != isinstance(theta, u.Quantity):
             raise TypeError("All arguments must be of the same type "
                             "(i.e., quantity or not).")
 
```

Comment:
On the other hand, shouldn't the test be explicit that the Quantity must have a specific angle consistent unit? Not just any quantity. True the code following will raise an error, but to be consistent with your desire for clear error messages, shouldn't angle consistent units be tested for? (Though I find it a bit weird that one could supply radians as a quantity, have it converted to degrees only to have it converted back to radians, but I suppose that is a matter of degree...

Reply:
@perrygreenfield I prefer to have it this way because I am mostly not familiar with `Quantity` (i.e., not a frequent user). I assumed angles could be defined not only in `deg` vs `rad` but also `arcsec`, hours, and whatnot. If one angle is in degrees and another one is in hours, should the code fail? If that is expected behavior, I will be happy to add this.